### PR TITLE
[SPARK-35035][PYTHON] Port Koalas internal implementation unit tests into PySpark

### PR DESCRIPTION
**Pull request by xinrong-meng**
### What changes were proposed in this pull request?
Now that we merged the Koalas main code into the PySpark code base (#32036), we should port the Koalas internal implementation unit tests to PySpark.

### Why are the changes needed?
Currently, the pandas-on-Spark modules are not tested fully. We should enable the internal implementation unit tests.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Enable internal implementation unit tests.

Keyword:SPARK-34849

### CODE REVIEW

**Comment by xinrong-databricks** [comment]
CC @ueshin @HyukjinKwon @itholic

**Comment by xinrong-databricks** [comment]
Conflicts are about the order of test names in the test module.
I'm not sure which ongoing PR will be merged first.
I would prefer to resolve conflicts when other parts look good.

File: dev/sparktestsupport/modules.py [hunk 1]
```diff
@@ -612,6 +612,13 @@ def __hash__(self):
         "pyspark.pandas.typedef.typehints",
         # unittests
         "pyspark.pandas.tests.test_dataframe",
+        "pyspark.pandas.tests.test_config",
```
Comment (xinrong-databricks) [inline dev/sparktestsupport/modules.py:615]: May I sort these tests in my last PR? That would make resolving other PRs' conflicts much easier.

**Review by ueshin — APPROVED** [review]
LGTM, pending tests.

**Review by HyukjinKwon — APPROVED** [review]

**Review by itholic — APPROVED** [review]
LGTM. Thanks!

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**